engine/swarm/swarm_manager_locking.md

```diff
@@ -0,0 +1,157 @@
+---
+description: Automatically lock Swarm managers to protect encryption keys
+keywords:
+- swarm, manager, lock, unlock, autolock, encryption
+title: Lock your swarm to protect its encryption key
+---
+
+In Docker 1.13 and higher, the RAFT logs used by swarm managers are encrypted on
+disk by default. This at-rest encryption protects your service's configuration
+and data from attackers who gain access to the encrypted RAFT logs. One of the
+reasons this feature was introduced was in support of the new [Docker
+secrets](secrets.md) feature.
+
+When Docker restarts, both the TLS key used to encrypt communication among swarm
+nodes, and the key used to encrypt and decrypt RAFT logs on disk, are loaded
+into each manager node's memory. Docker 1.13 introduces the ability to protect
+the mutual TLS encryption key and the key used to encrypt and decrypt RAFT logs
+at rest, by allowing you to take ownership of this key and to require manual
+unlocking of your managers. This feature is called _autolock_.
+
+When Docker restarts, you must
+[unlock the swarm](swarm_manager_locking.md#unlock-a-swarm) first, using a key
+generated by Docker when the swarm was locked. You can rotate this key at any
+time.
+
+>**Note**: You don't need to unlock the swarm when a new node joins the swarm,
+because the key is propagated to it over mutual TLS.
+
+## Initialize a swarm with autolocking enabled
+
+When you initialize a new swarm, you you can use the `--autolock` flag to
+enable autolocking of swarm manager nodes when Docker restarts.
+
+```bash
+$ docker swarm init --autolock
+
+Swarm initialized: current node (k1q27tfyx9rncpixhk69sa61v) is now a manager.
+
+To add a worker to this swarm, run the following command:
+
+    docker swarm join \
+    --token SWMTKN-1-0j52ln6hxjpxk2wgk917abcnxywj3xed0y8vi1e5m9t3uttrtu-7bnxvvlz2mrcpfonjuztmtts9 \
+    172.31.46.109:2377
+
+To add a manager to this swarm, run 'docker swarm join-token manager' and follow the instructions.
+
+To unlock a swarm manager after it restarts, run the `docker swarm unlock`
+command and provide the following key:
+
+    SWMKEY-1-WuYH/IX284+lRcXuoVf38viIDK3HJEKY13MIHX+tTt8
+```
+
+Store the key in a safe place, such as in a password manager.
+
+When Docker restarts, you need to
+[unlock the swarm](swarm_manager_locking.md#unlock-a-swarm). You will see an
+error like the following and services will not start.
+
+```bash
+$ sudo service docker restart
+
+$ docker service ls
+
+Error response from daemon: Swarm is encrypted and needs to be unlocked before it can be used. Please use "docker swarm unlock" to unlock it.
+```
+
+## Enable or disable autolock on an existing swarm
+
+To enable autolock on an existing swarm, set the `autolock` flag to `true`.
+
+```bash
+$ docker swarm update --autolock="true"
```

Comment:
It's technically valid, but it's very atypical to have double quotes there. You could also write it as `docker "swarm" "update" "--autolock"=true`, but people don't do that either. So it looks like there are quotation marks in an arbitrary place where they do nothing, and I think it would look strange to people who understand the shell's syntax rules.